PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp

```diff
@@ -1208,10 +1787,246 @@ int FileRelocalizerDlg::doDialog(const TCHAR *fn, bool isRTL)
+			goToCenter();
+			return TRUE;
+		}
+		case WM_COMMAND : 
+		{
+			switch (wParam)
+			{
+				case IDOK :
+				{
+					_filters.clear();
+					const int filterSize = 256;
+					TCHAR tfilters[filterSize + 1];
+					tfilters[filterSize] = '\0';
+					::GetDlgItemText(_hSelf, IDD_PROJECTPANEL_FILTERS_COMBO, tfilters, filterSize);
+					addText2Combo(tfilters, ::GetDlgItem(_hSelf, IDD_PROJECTPANEL_FILTERS_COMBO));
+
+					_filters = stringSplit(tfilters, TEXT(";"), true);
+
+					// strange behaviour of stringSplit - it returns an array with a single empty string if it gets an empty string.
+					if (_filters.size() == 1 && _filters[0].empty())
+						_filters.clear();
+
+					// check for nonsense combinations with *.* (which removes all filters)
+					for (size_t i=0; i<_filters.size(); ++i)
+					{
+						if (_filters[i] == TEXT("*.*"))
+						{
+							_filters.clear();
+							break;
+						}
+					}
+
+					NppParameters *nppParams = NppParameters::getInstance();
+					FindHistory & findHistory = nppParams->getFindHistory();
+					saveComboHistory(IDD_PROJECTPANEL_FILTERS_COMBO, findHistory._nbMaxFindHistoryFilter, findHistory._findHistoryFilters);
+
+					::EndDialog(_hSelf, 0);
+				}
+				return TRUE;
+
+				case IDCANCEL :
+					::EndDialog(_hSelf, -1);
+				return TRUE;
+
+				default:
+					return FALSE;
+			}
+		}
+		default :
+			return FALSE;
+	}
+}
+
+void FilterDlg::addText2Combo(const TCHAR * txt2add, HWND hCombo)
+{
+	if (!hCombo) return;
+	if (!lstrcmp(txt2add, TEXT(""))) return;
+
+	int i = 0;
+
+	i = ::SendMessage(hCombo, CB_FINDSTRINGEXACT, static_cast<WPARAM>(-1), reinterpret_cast<LPARAM>(txt2add));
+	if (i != CB_ERR) // found
+	{
+		::SendMessage(hCombo, CB_DELETESTRING, i, 0);
+	}
+
+	i = ::SendMessage(hCombo, CB_INSERTSTRING, 0, reinterpret_cast<LPARAM>(txt2add));
+
+	::SendMessage(hCombo, CB_SETCURSEL, i, 0);
+}
+
+generic_string FilterDlg::getTextFromCombo(HWND hCombo, bool) const
+{
+	TCHAR str[PROJECTPANEL_FILTERS_MAXLENGTH];
+	::SendMessage(hCombo, WM_GETTEXT, PROJECTPANEL_FILTERS_MAXLENGTH - 1, reinterpret_cast<LPARAM>(str));
+	return generic_string(str);
+}
+
+
+
+void FilterDlg::fillComboHistory(int id, const std::vector<generic_string> & strings)
+{
+	HWND hCombo = ::GetDlgItem(_hSelf, id);
+
+	for (std::vector<generic_string>::const_reverse_iterator i = strings.rbegin(); i != strings.rend(); ++i)
+	{
+		addText2Combo(i->c_str(), hCombo);
+	}
+	::SendMessage(hCombo, CB_SETCURSEL, 0, 0); // select first item
+}
+
+int FilterDlg::saveComboHistory(int id, int maxcount, std::vector<generic_string> & strings)
+{
+	TCHAR text[PROJECTPANEL_FILTERS_MAXLENGTH];
+	HWND hCombo = ::GetDlgItem(_hSelf, id);
+	int count = ::SendMessage(hCombo, CB_GETCOUNT, 0, 0);
+	count = min(count, maxcount);
+
+	if (count == CB_ERR) return 0;
+
+	if (count)
+		strings.clear();
+
+	for (int i = 0; i < count; ++i)
+	{
+		::SendMessage(hCombo, CB_GETLBTEXT, i, reinterpret_cast<LPARAM>(text));
+		strings.emplace_back(generic_string(text));

```

Comment:
From the [documentation](https://msdn.microsoft.com/en-us/library/windows/desktop/bb775862.aspx):

> Security Warning:  Using this message incorrectly can compromise the security of your program. This message does not provide a way for you to know the size of the buffer. If you use this message, first call CB_GETLBTEXTLEN to get the number of characters that are required and then call the message to retrieve the string. You should review the Security Considerations: Microsoft Windows Controls before continuing.

You should use `CB_GETLBTEXTLEN` ***instead of*** `CB_GETCOUNT`.